src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/vilim2009/PropCumulative.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.vilim2009;
+
+import org.chocosolver.solver.constraints.nary.cumulative.literature.CumulativeFilter;
+import gnu.trove.list.array.TIntArrayList;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.Comparator;
+import java.util.HashMap;
+
+/**
+ * Cumulative constraint filtering algorithms described in the following paper :
+ * Vilim, P.: Edge finding filtering algorithm for discrete cumulative resources in O(k n log(n)). In: Proceedings of the 15th International Conference on Principles and Practice of Constraint Programming (CP 2009), pp. 802-816 (2009). https://doi.org/10.1007/978-3-642-04244-7_62
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropCumulative extends CumulativeFilter {
```

Comment:
The class name should not be so generic, there are many other `PropCumulative` classes, that may be confusing.